server/runtime_nakama_module.go

```diff
@@ -943,20 +943,21 @@ func (n *NakamaModule) groupsCreate(l *lua.LState) int {
 				}
 				p.Private = lua.LVAsBool(v)
 			case "Metadata":
-				if v.Type() != lua.LTString {
+				if v.Type() != lua.LTTable {
 					conversionError = true
-					l.ArgError(1, "expects Metadata to be string")
+					l.ArgError(1, "expects Metadata to be a table")
 					return
 				}
 
-				j := []byte(v.String())
-				maybeJson := make(map[string]interface{})
-				if jsonErr := json.Unmarshal(j, &maybeJson); jsonErr != nil {
+				metadataMap := ConvertLuaTable(v.(*lua.LTable))
+				metadataBytes, err := json.Marshal(metadataMap)
```

Comment:
I don't like this. For instance, in the case of GroupCreate, the proto already expects the `metadata` field to be Bytes. No point putting this extra conversion in the pipeline.